.circleci/soltest_all.sh

```diff
@@ -28,10 +28,46 @@ set -e
 
 REPODIR="$(realpath $(dirname $0)/..)"
 
-EVM=istanbul OPTIMIZE=1 ABI_ENCODER_V2=1 ${REPODIR}/.circleci/soltest.sh
+# This number has to equal the number of steps below.
+STEPS=11
+
+if [ "$CIRCLE_NODE_TOTAL" -a "$CIRCLE_NODE_TOTAL" -gt 1 ]
+then
+    # Run step 1 as the only step on the first executor
+    # and evenly distribute the other steps among
+    # the other executors.
+    # The first step takes much longer than the other steps.
+    if [ "$CIRCLE_NODE_INDEX" = "0" ]
```

Comment:
Arithmetic expressions would be more readable and would not require quoting stuff inside them:
```suggestion
if (( $CIRCLE_NODE_TOTAL )) && (( $CIRCLE_NODE_TOTAL > 1 ))
then
    # Run step 1 as the only step on the first executor
    # and evenly distribute the other steps among
    # the other executors.
    # The first step takes much longer than the other steps.
    if (( $CIRCLE_NODE_INDEX == 0 ))
```